### PR TITLE
[CPDLP-3693] Data cleanup - lookup application directly by id in decl…

### PR DESCRIPTION
**Pull request by mooktakim**
…aration migrator

### Context

Ticket: https://dfedigital.atlassian.net/browse/CPDLP-3693

### Changes proposed in this pull request

* Use `ecf_id` from `participant_profile.npq_application.id` directly for declarations.

### CODE REVIEW

File: spec/services/migration/migrators/declaration_spec.rb [hunk 1]
```diff
@@ -32,8 +32,7 @@ def setup_failure_state
         expect(declaration.state_reason).to eq(ecf_resource1.declaration_states.last.state_reason)
         expect(declaration.cohort.start_year).to eq(ecf_resource1.cohort.start_year)
         expect(declaration.lead_provider.ecf_id).to eq(ecf_resource1.cpd_lead_provider.npq_lead_provider.id)
-        expect(declaration.application.course.identifier.downcase).to eq(ecf_resource1.course_identifier.downcase)
```
Comment (cwrw) [inline spec/services/migration/migrators/declaration_spec.rb:35]: not sure why we removed those tests, they should still hold 🤔

**Review by cwrw — APPROVED** [review]
just one question

File: spec/services/migration/migrators/declaration_spec.rb (hunk 1 above)
Comment (mooktakim) [reply spec/services/migration/migrators/declaration_spec.rb:35]: I thought they weren't need anymore as we're testing if application is correct. 
I've changed it back now.

**Review by leandroalemao — APPROVED** [review]

**Comment by mooktakim** [comment]
Tested on migration env

**Pull request merged**